hyp3_gamma/dem.py

```diff
@@ -110,5 +128,10 @@ def prepare_dem_geotiff(output_name: str, geometry: ogr.Geometry, pixel_size: fl
             gdal.BuildVRT(str(dem_vrt), dem_file_paths)
 
             epsg_code = utm_from_lon_lat(centroid.GetX(), centroid.GetY())
-            gdal.Warp(output_name, str(dem_vrt), dstSRS=f'EPSG:{epsg_code}', xRes=pixel_size, yRes=pixel_size,
+
+            tmp_tif = temp_path / 'tmp.tif'
+            gdal.Warp(str(tmp_tif), str(dem_vrt), dstSRS=f'EPSG:{epsg_code}', xRes=pixel_size, yRes=pixel_size,
                       targetAlignedPixels=True, resampleAlg='cubic', multithread=True)
+
+            nodataval = min_value_datatype(dem_file_paths[0])
+            gdal.Translate(output_name, str(tmp_tif), noData=nodataval)
```

Comment:
There may not be a need for the separate `gdal.Translate` call to assign the new NoData value, I think we might get the same results by simply setting the `dstNodata=nodataval` option in the `gdal.Warp` call

Reply:
I tested gdal.warp with option dstNodata=nodataval, it does not work.  This is why I use separate  gdal.Translate to do it.